src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -43,14 +44,85 @@ public partial class GenericVectorTests
         private void TestConstructorWithSpan<T>() where T : struct
         {
             T[] values = GenerateRandomValuesForVector<T>().ToArray();
-            var valueSpan = new Span<T>(values);
+            Vector<T> vector;
 
-            var vector = new Vector<T>(valueSpan);
+            // Span<T> ctor
+            vector = new Vector<T>(new Span<T>(values));
             ValidateVector(vector,
                 (index, val) =>
                 {
                     Assert.Equal(values[index], val);
                 });
+
+            // ReadOnlySpan<T> ctor
+            vector = new Vector<T>(new ReadOnlySpan<T>(values));
+            ValidateVector(vector,
+                (index, val) =>
+                {
+                    Assert.Equal(values[index], val);
+                });
+
+            // ReadOnlySpan<byte> ctor
+            vector = new Vector<T>(MemoryMarshal.AsBytes(new ReadOnlySpan<T>(values)));
+            ValidateVector(vector,
+                (index, val) =>
+                {
+                    Assert.Equal(values[index], val);
+                });
+        }
+
+        [Fact]
+        public void ReadOnlySpanBasedConstructorWithLessElements_Byte() => Assert.Throws<IndexOutOfRangeException>(() => TestReadOnlySpanBasedConstructorWithLessElements<byte>());
```

Comment:
Seems like having a customized `Theory` attribute that took a set of `T` as part of the `InlineData` would be useful...

Reply:
(to clarify, not saying this needs to be done here/now; just that it might be useful in the future)